include/svs/index/vamana/iterator.h

```diff
@@ -47,181 +50,145 @@ template <std::integral I> struct RestartInitializer {
+    void copy_from_scratch(size_t batch_size) {
         results_.clear();
         const auto& buffer = scratchspace_.buffer;
         for (size_t i = 0, imax = buffer.size(); i < imax; ++i) {
             auto neighbor = buffer[i];
-            // Unfortunately, lambdas cannot capture structured bindings.
-            // So we need to manually unpack the result from `insert`.
             auto result = yielded_.insert(neighbor.id());
-            if (result.second /*inserted*/) {
-                // If push_back throws - rollback the insertion into the `yielded` set.
+            if (result.second /* inserted */) {
+                // Rollback insertion into the yielded set if push_back throws.
                 auto guard = lib::make_dismissable_scope_guard([&]() noexcept {
-                    // `std::unordered_set` documented to not throw exceptions on `erase`.
                     yielded_.erase(result.first);
                 });
                 results_.push_back(adapt(neighbor));
                 guard.dismiss();
             }
 
-            // Break if we've yielded the requested number of candidates.
-            if (results_.size() == max_candidates) {
+            // Stop if the requested batch size is reached.
+            if (results_.size() == batch_size) {
                 break;
             }
         }
     }
 
+    /// @brief Initializes the search buffer with the configured capacity.
+    void initialize_buffer() {
+        auto config = SearchBufferConfig{0, extra_search_buffer_capacity_};
+        scratchspace_.buffer.change_maxsize(config);
+    }
+
+    /// @brief Increments the search window and capacity by `batch_size` for the next
+    /// iteration.
+    void increment_buffer(size_t batch_size) {
+        auto config = scratchspace_.buffer.config();
+        config.increment(batch_size);
+        scratchspace_.buffer.change_maxsize(config);
+    }
+
   public:
     using size_type = typename result_buffer_type::size_type;
     using reference = value_type&;
     using const_reference = const value_type&;
 
-    /// A random-access, contiguous iterator to ``value_type`` over the current batch of
-    /// results.
+    /// Random-access iterator to `value_type` over the current batch of results.
     using iterator = typename result_buffer_type::iterator;
-    /// A random-access, contiguous iterator to ``cosnt value_type`` over the current batch
-    /// of results.
+    /// Random-access iterator to `const value_type` over the current batch of results.
     using const_iterator = typename result_buffer_type::const_iterator;
 
+    /// @brief Constructs a batch iterator for the given query over the index.
+    /// @param parent The index to search.
+    /// @param query The query data.
+    /// @param extra_search_buffer_capacity Additional buffer capacity for the search
+    ///     When not provided, default value SVS_ITERATOR_EXTRA_BUFFER_CAPACITY_DEFAULT is
```

Comment:
I think `SVS_ITERATOR_EXTRA_BUFFER_CAPACITY_DEFAULT` is used literally, not replaced by 100 by Doxygen. If we can make the definition of `SVS_ITERATOR_EXTRA_BUFFER_CAPACITY_DEFAULT` show up in the HTML docs, that's great. If not, we should just use `100` here.

In any case the same solution for presenting the default should be used here and in the `batch_iterator` methods.

Reply:
I gave it a shot, but pulling the values of these global variables/defines in C++ with Doxygen/Sphinx isn't as simple as generating Python docs. I'd rather stick with using the variable name instead of hardcoded values in docs, since hardcoded values can easily become outdated or inconsistent from the source code

Reply:
I agree, it is better to not hard code the value in the docs. Nevertheless, we should show the value in the docs; users should not have to look at the source code to learn that value is 100. Looking at the HTML docs, I don't see 100.

Reply:
Yeah, created issue #118 to properly get the values of these globals